code/obj/artifacts/artifact_objects/borg_maker.dm

```diff
@@ -24,19 +32,49 @@
 		if (converting)
 			return
 		if (ishuman(user))
+			var/mob/living/carbon/human/humanuser = user
 			if(!isalive(user) && user.ghost && user.ghost.mind && user.ghost.mind.dnr)
 				O.visible_message("<span class='alert'><b>[O]</b> refuses to process [user.name]!</span>")
 				return
-			O.visible_message("<span class='alert'><b>[O]</b> suddenly pulls [user.name] inside and slams shut!</span>")
+			O.visible_message("<span class='alert'><b>[O]</b> suddenly pulls [user.name] inside[escapable ? "!" : " and slams shut!"]</span>")
 			user.emote("scream")
-			user.set_loc(O.loc)
-			converting = 1
-			var/loops = rand(10,20)
+			if (!escapable)
+				user.set_loc(O)
+			else
+				user.set_loc(get_turf(O.loc))
+			converting = TRUE
+			var/list/obj/item/parts/convertable_limbs = list(humanuser.limbs.l_arm, humanuser.limbs.r_arm, humanuser.limbs.l_leg, humanuser.limbs.r_leg)
+			//figure out which limbs are already robotic and remove them from the list
+			for (var/obj/item/parts/limb in convertable_limbs)
+				if (!limb || (limb.kind_of_limb & LIMB_ROBOT))
+					convertable_limbs -= limb
+			//people with existing robolimbs get converted faster.
+			//(loops_per_conversion_step - 1) bit adds some 'buffer time' before any limbs are converted.
+			var/loops = (loops_per_conversion_step * (convertable_limbs.len + 1)) + (loops_per_conversion_step - 1)
 			while (loops > 0)
+				if (user.loc != O.loc && user.loc != O)
```

Comment:
Doesn't really work great with the artifact being deactivated (for instance via activator key), it just keeps going.
Maybe consider adding something like this to make it stop?
```suggestion
				if ((user.loc != O.loc && user.loc != O) || !src.activated)
```

In that case, however, the inescapable type would leave them stuck inside the artifact.
You could make more code to release everyone inside when deactivating.

People can also just damage the artifact they are inside of by flipping until it is destroyed, but I think a lot of people aren't aware of that, so it's your call if you want to leave them inside or eject them.

Apart from that minor issue, it works great in my testing!

Reply:
Ah, I totally forgot activator keys could do that. I think trapping people inside would create interesting situations, but you're right that not many people know about the flip-to-break thing; in fact, most people I see don't know how to break artifacts at all. For now it just ejects people on deactivation.